app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/JunkBottomSheetDialog.kt

```diff
@@ -58,15 +60,32 @@ class JunkBottomSheetDialog : ActionsBottomSheetDialog() {
         }
 
         observeReportPhishingResult()
+        observeExpeditorsResult(threadUid)
     }
 
-    fun observeReportPhishingResult() {
+    private fun observeReportPhishingResult() {
         mainViewModel.reportPhishingTrigger.observe(viewLifecycleOwner) {
             descriptionDialog.resetLoadingAndDismiss()
             findNavController().popBackStack()
         }
     }
 
+    private fun observeExpeditorsResult(threadUid: String) = with(binding) {
+        mainViewModel.hasOtherExpeditors(threadUid).observe(viewLifecycleOwner) { hasOtherExpeditors ->
+            if (hasOtherExpeditors) {
+                blockSender.setClosingOnClickListener {
+                    safeNavigate(
+                        resId = R.id.userToBlockBottomSheetDialog,
+                        args = UserToBlockBottomSheetDialogArgs(threadUid).toBundle(),
+                        currentClassName = JunkBottomSheetDialog::class.java.name,
+                    )
+                }
+            } else {
+                blockSender.isGone = true
```

Comment:
Shouldn't we instead remove the possibility to go to the "Report junk" bottomSheet in this case ?

Reply:
After discussing with @FabianDevel, we realized that we had already addressed this issue a long time ago. We decided to avoid handling multiple cases in this BottomSheet (in a Thread, in a Message, etc.). Webmail still allows blocking a thread where you're the only contributor, so perhaps we should maintain this functionality, even though it seems counterintuitive ...